cpp/ql/lib/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -131,16 +131,23 @@ private class IteratorPointerDereferenceOperator extends Operator, TaintFunction
 /**
  * A non-member `operator++` or `operator--` function for an iterator type.
  */
-private class IteratorCrementOperator extends Operator, DataFlowFunction {
+class IteratorCrementOperator extends Operator {
   FunctionInput iteratorInput;
 
   IteratorCrementOperator() {
     this.hasName(["operator++", "operator--"]) and
     iteratorInput = getIteratorArgumentInput(this, 0)
   }
 
+  /**
+   * INTERNAL: Do not use.
```

Comment:
Is there a strong reason for having this here rather than in `IteratorCrementOperatorModel`, and just having `exists(getIteratorArgumentInput(this, 0))` in the constructor here if necessary?

`IteratorAssignArithmeticOperator` is already designed this way.

Reply:
No, we could do that if you prefer. I mainly did it because `IteratorCrementOperator` already had to check for the presence of an iterator input argument, and so I chose to implement it here to increase code sharing.

Reply:
I'd prefer it to be properly private tucked away in `IteratorCrementOperatorModel`.  I don't think the code sharing benefit is very substantial.  Though if there might be performance benefits as well I can live with it being here (as long as all of them have the `INTERNAL` comment of course).

Reply:
Good point. I've removed these member predicates in 7583fe2ad850773fd3939b36ff9292dabbf6a7c6.